lib/spack/docs/packaging_guide.rst

```diff
@@ -931,77 +964,117 @@ checksum.
+     class Example(Package):
+
+         hg = "https://bitbucket.org/example-project/example"
 
-  This download method is untrusted, and is not recommended.
+         version('develop')
+
+  This download method is untrusted, and is not recommended. As with
+  Git's default fetching strategy, there is no way to verify the
+  integrity of the download.
 
 Revisions
-  Add ``hg`` and ``revision`` parameters:
+  To fetch a particular revision, use the ``revision`` parameter:
 
   .. code-block:: python
 
-     version('1.0', hg='https://jay.grs.rwth-aachen.de/hg/example',
-             revision='v1.0')
-
-  This download method is untrusted, and is not recommended.
+     version('1.0', revision='v1.0')
 
   Unlike ``git``, which has special parameters for different types of
   revisions, you can use ``revision`` for branches, tags, and commits
-  when you fetch with Mercurial.
+  when you fetch with Mercurial. Like Git, fetching specific branches
+  or tags is an untrusted download method, and is not recommended.
+  The recommended fetch strategy is to specify a particular commit
+  hash as the revision.
 
-As with git, you can fetch these versions using the ``spack install
-example@<version>`` command-line syntax.
 
 .. _svn-fetch:
 
 ^^^^^^^^^^
 Subversion
 ^^^^^^^^^^
 
-To fetch with subversion, use the ``svn`` and ``revision`` parameters:
+To fetch with subversion, use the ``svn`` and ``revision`` parameters.
 
 Fetching the head
-  Simply add an ``svn`` parameter to ``version``:
+  Simply add an ``svn`` parameter to the package:
 
   .. code-block:: python
 
-     version('develop', svn='https://outreach.scidac.gov/svn/libmonitor/trunk')
+     class Example(Package):
+
+         svn = "https://outreach.scidac.gov/svn/example/trunk"
 
-  This download method is untrusted, and is not recommended.
+         version('develop')
+
+  This download method is untrusted, and is not recommended for the
+  same reasons as mentioned above.
 
 Fetching a revision
-  To fetch a particular revision, add a ``revision`` to the
-  version call:
+  To fetch a particular revision, add a ``revision`` argument to the
+  version directive:
 
   .. code-block:: python
 
-     version('develop', svn='https://outreach.scidac.gov/svn/libmonitor/trunk',
-             revision=128)
+     version('develop', revision=128)
 
   This download method is untrusted, and is not recommended.
 
+  Unfortunately, Subversion has no commit hashing scheme like Git and
+  Mercurial do, so there is no way to guarantee that the download you
+  get is the same as the download used when the package was created.
+  Use at your own risk.
+
 Subversion branches are handled as part of the directory structure, so
-you can check out a branch or tag by changing the ``url``.
+you can check out a branch or tag by changing the URL. If you want to
+package multiple branches, simply add a ``svn`` argument to each
+version directive.
+
+.. _go-fetch:
+
+^^
+Go
```

Comment:
Apparently Spack supports fetching with Go? Well, it's documented now! Tbh, I don't know the first thing about Go, so if this section feels... wishy-washy, that's why. Someone should definitely take a close look at this section and correct me wherever I'm wrong.